### PR TITLE
eng, fix typespec compare script

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
# Description

- After this fix, CI will fail, if there is diff in current code and `tsp compile` code

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/scripts/TypeSpec-Compare-CurrentToCodegeneration.ps1 [hunk 1]
```diff
@@ -51,7 +51,7 @@ if ($LastExitCode -ne 0) {
 The following files are out of date:
 $status
 "
-  exit $LASTEXITCODE
+  exit 1
```
Comment (weidongxu-microsoft) [inline eng/scripts/TypeSpec-Compare-CurrentToCodegeneration.ps1:54]: `LASTEXITCODE` is code from the `Write-Host`, which is always 0.

This causes the script always return success, even if there is diff. And then error could happen (or not, depends on the diff) in later steps. e.g. https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2937216&view=logs&jobId=b70e5e73-bbb6-5567-0939-8415943fadb9&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=d31d5e2d-d8df-53bc-da94-87e19f5b2ee4

Comment (weshaggard) [reply eng/scripts/TypeSpec-Compare-CurrentToCodegeneration.ps1:54]: It isn't the return code from Write-Host as cmdlets don't set it but it is the return code of the git status call. So if we want to return the same code we need to capture it before the git status call but I think returning 1 here is probably fine as well.

**Review by weshaggard — APPROVED** [review]

File: eng/scripts/TypeSpec-Compare-CurrentToCodegeneration.ps1 (hunk 1 above)
Comment (benbp) [reply eng/scripts/TypeSpec-Compare-CurrentToCodegeneration.ps1:54]:
```
$ git asdfasdfasdfas
git: 'asdfasdfasdfas' is not a git command. See 'git --help'.
$ $LASTEXITCODE
1
$ write-host foo
foo
$ $LASTEXITCODE
1
```

Comment (weidongxu-microsoft) [reply eng/scripts/TypeSpec-Compare-CurrentToCodegeneration.ps1:54]: Thanks Wes and Ben! Learnt it.

**Pull request merged**